src/main/scala/com/raphtory/core/config/PulsarController.scala

```diff
@@ -78,36 +78,60 @@ class PulsarController(conf: Config) {
     pulsarAdmin.namespaces().setDeduplicationStatus(namespace, false)
   }
 
-  def createListeningConsumer[T](
+  private def configuredConsumerBuilder[T](
       subscriptionName: String,
-      messageListener: MessageListener[T],
       schema: Schema[T],
-      topics: String*
-  ): Consumer[T] =
+      topics: Seq[String]
+  ): ConsumerBuilder[T] =
     client
       .newConsumer(schema)
       .priorityLevel(0)
       .topics(topics.toList.asJava)
       .subscriptionName(subscriptionName)
-      .subscriptionType(SubscriptionType.Shared)
+      .subscriptionType(SubscriptionType.Exclusive)
       .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
       .maxTotalReceiverQueueSizeAcrossPartitions(Integer.MAX_VALUE)
       .receiverQueueSize(200_000)
       .poolMessages(true)
+
+  def createSharedListeningConsumer[T](
+      subscriptionName: String,
+      messageListener: MessageListener[T],
+      schema: Schema[T],
+      topics: String*
+  ): Consumer[T] =
+    configuredConsumerBuilder(subscriptionName, schema, topics)
       .messageListener(messageListener)
+      .subscriptionType(SubscriptionType.Shared)
       .subscribe()
 
-  def createConsumer[T](subscriptionName: String, schema: Schema[T], topics: String*): Consumer[T] =
-    client
-      .newConsumer(schema)
-      .priorityLevel(0)
-      .topics(topics.toList.asJava)
-      .subscriptionName(subscriptionName)
+  def createExclusiveListeningConsumer[T](
+      subscriptionName: String,
+      messageListener: MessageListener[T],
+      schema: Schema[T],
+      topics: String*
+  ): Consumer[T] =
+    configuredConsumerBuilder(subscriptionName, schema, topics)
+      .messageListener(messageListener)
+      .subscriptionType(SubscriptionType.Exclusive)
+      .subscribe()
+
+  def createSharedConsumer[T](
+      subscriptionName: String,
+      schema: Schema[T],
+      topics: String*
+  ): Consumer[T] =
+    configuredConsumerBuilder(subscriptionName, schema, topics)
       .subscriptionType(SubscriptionType.Shared)
-      .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
-      .maxTotalReceiverQueueSizeAcrossPartitions(Integer.MAX_VALUE)
-      .receiverQueueSize(200_000)
-      .poolMessages(true)
+      .subscribe()
+
+  def createExclusiveConsumer[T](
```

Comment:
No reason not to have it, but this isn't used anywhere at the moment is it?

Reply:
Not used but felt very strange not to have it here...